WireAPI/Sources/WireAPI/Private/APIs/BackendInfoAPI/BackendInfoAPIV0.swift

```diff
@@ -0,0 +1,69 @@
+//
+// Wire
+// Copyright (C) 2024 Wire Swiss GmbH
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with this program. If not, see http://www.gnu.org/licenses/.
+//
+
+import Foundation
+
+class BackendInfoAPIV0: BackendInfoAPI {
+
+    let httpClient: HTTPClient
+
+    init(httpClient: HTTPClient) {
```

Comment:
It looks upside down to me to pass the http client into the endpoint description. I would expect to have a http client that is fed with endpoint descriptions and receives a certain result (`Data`?) from the server and decodes the result into a model.

Reply:
Can you provide a code example?

Reply:
I've added another package next to this one to compare: https://github.com/wireapp/wire-ios/pull/1309